### PR TITLE
Regional base price selection

### PR DESCRIPTION
**Pull request by gino247**
Addresses the difference in base price.
Also started introducing sinon, for proper util stubing.
Removed minCost environment variable since it is not needed any longer.
#77

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (alexcasalboni) [inline .vscode/settings.json:1]: Can you please exclude this file from the PR?

File: template.yml [hunk 2]
```diff
@@ -39,7 +39,7 @@ Globals:
       Variables:
         defaultPowerValues: !Join [ ",", !Ref PowerValues ]
         minRAM: '128'
-        minCost: '0.000000208'
+        minCosts: 'ap-east-1=0.0000002865,af-south-1=0.0000002763,me-south-1=0.0000002583,default=0.000000208'
```
Comment (alexcasalboni) [inline template.yml:42]: Why not a JSON instead? The implementation of `generateBasePriceMap` would be much simpler and elegant.

File: lambda/executor.js [hunk 3]
```diff
@@ -15,6 +15,10 @@ module.exports.handler = async(event, context) => {
 
     validateInput(lambdaARN, value, num); // may throw
 
+    // get minCost map and min cost
+    const minCostsMap = generateBasePriceMap(minCosts, lambdaARN)
```
Comment (alexcasalboni) [inline lambda/executor.js:19]: I guess you don't need to pass `lambdaARN` as second argument here? Since it doesn't depend on the input event, this env variable parsing could happen out of the handler?

File: test/test-lambda.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
 'use strict';
 
+const sinon = require('sinon');
```
Comment (alexcasalboni) [inline test/test-lambda.js:3]: This is great stuff 🚀It'd be great if we could open another PR only with this sinon integration, just to keep the two independent. No worries if it's too much trouble, but i'd appreciate it.

File: template.yml (hunk 2 above)
Comment (alexcasalboni) [inline template.yml:42]: Also, can you please integrate #76 here? The default base price should be `0.0000002083`.

File: .vscode/settings.json (hunk 1 above)
Comment (gino247) [reply .vscode/settings.json:1]: Thought I removed it all.
Will do.

File: template.yml (hunk 2 above)
Comment (gino247) [reply template.yml:42]: Would you like the json as a env or as a seperate file that is used as:
`const prices = require('./prices.json');`?

File: lambda/executor.js (hunk 3 above)
Comment (gino247) [reply lambda/executor.js:19]: The implementation needs to know in which region this lambda is running to return the appropriate cost.

File: test/test-lambda.js (hunk 4 above)
Comment (gino247) [reply test/test-lambda.js:3]: I was having issues with the testing.
So I started using sinon and then it took over.
Will it be possible to rather start a now request for testing improvements?

File: template.yml (hunk 2 above)
Comment (gino247) [reply template.yml:42]: Will do.

Comment (alexcasalboni) [reply template.yml]: I was thinking of a JSON string in the environment variable.

File: lambda/executor.js (hunk 3 above)
Comment (alexcasalboni) [reply lambda/executor.js:19]: But `generateBasePriceMap` only has one input parameter :)

File: test/test-lambda.js (hunk 4 above)
Comment (alexcasalboni) [reply test/test-lambda.js:3]: No worries then, I understand. Let's keep it here. Then yes, let's create a new issue & PR for testing improvements once this is merged.

File: .vscode/settings.json (hunk 1 above)
Comment (gino247) [reply .vscode/settings.json]: Removed.

File: template.yml (hunk 2 above)
Comment (gino247) [reply template.yml]: Cool. Made changes to address this.
Renamed variable to `baseCosts`.

File: lambda/executor.js (hunk 3 above)
Comment (gino247) [reply lambda/executor.js]: Hahahaha.
My bad.
Rewrote this a bit please check.

File: template.yml (hunk 2 above)
Comment (gino247) [reply template.yml]: Sorted.

File: test/test-lambda.js (hunk 4 above)
Comment (gino247) [reply test/test-lambda.js:3]: If all other changes are good.
Give me a day or two and maybe I can revert the tests and then create a new PR for the tests.
Added a simple why to suppress the logs during tests runs.

Comment (gino247) [reply test/test-lambda.js]: Ok was easier than I thought.
Reverted the changes.
Will create a separate PR with the test stuff.

**Comment by alexcasalboni** [comment]
This is awesome, thanks for the quick cleanup 🚀 

I'm going to add Milan (eu-south-1) too, which was announced last week with a base cost of `$0.0000002440`.

Now, the only missing piece left is the Step Functions price ^_^
I've already opened another issue: #80 

Do you feel like working on that one as well? The implementation will be quite similar :)

**Comment by gino247** [comment]
> This is awesome, thanks for the quick cleanup 🚀
> 
> I'm going to add Milan (eu-south-1) too, which was announced last week with a base cost of `$0.0000002440`.
> 
> Now, the only missing piece left is the Step Functions price ^_^
> I've already opened another issue: #80
> 
> Do you feel like working on that one as well? The implementation will be quite similar :)

Hi,

It's a pleasure.
Let's see what I can do, might only be on the weekend again.

Regards,
Eugene

**Comment by alexcasalboni** [comment]
Super!

Merging this PR now!

I'll wait for #2 and #80 to be implemented before publishing a new SAR app (see [milestone](https://github.com/alexcasalboni/aws-lambda-power-tuning/milestone/1)).

**Comment by gino247** [comment]
PS: Just saw a **;** in the **template.yml** at the end of the baseCost environment variable.

**Comment by alexcasalboni** [comment]
Thanks! I've fixed directly in `master` too.
